java-logging-appinsights/src/main/java/uk/gov/hmcts/reform/logging/appinsights/SpringApplicationName.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.logging.appinsights;
+
+import com.microsoft.applicationinsights.internal.util.PropertyHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * This is a hack we need until Microsoft delivers proper support for Spring applications. Telemetry initializers
+ * defined in ApplicationInsights.xml are not instrumented by Spring, we need to expose the application name via
+ * a static method in order to access it.
+ */
+public final class SpringApplicationName {
+
+    private static final String APPLICATION_NAME_PROPERTIES = "app-name.properties";
```

Comment:
- I don't think any components should be introduced other than from library itself
  - that being said, additional configs and tests that live in module right now could be moved to away to keep this clean
- Keeping in mind only Microsoft's library, this is elegant enough way (used in other initialisers too. mentioned in description) to load up additional properties wanted to be assigned to context
- Value is context value which can be assigned once per application lifetime per client. Some of them are invoked long before beans get initialised hence the huge possibility of having that context key empty
- There's no harm in keeping the name as is in projects which already implement this module. Once the spring boot starter becomes available if it ever will, then library can be adapted

I think these are my pros. The only con I can think of is having that extra file. Which to be honest should've been introduced from the very start. Since from init version only needed inside the bean creation, made only sense to keep in spring framework config instead